app/models/address.rb

```diff
@@ -30,18 +31,46 @@ def full_address
 
   alias to_s full_address
 
+  def geocode_with_ward
+    # TODO? Is there any point using the Geocoder gem rather than writing
+    # our own code to query postcodes.io given that using postcodes.io
+    # now defines how we use Geocoder?
+    # - Passing anything other than a postcode will cause Geocoder.search to
+    #   return an empty array.
+    # - We are accessing admin_ward directly through through
+    #   Geocoder::Result::PostcodesIo#data hash.
+
+    geo = Geocoder.search(postcode).first&.data
+
+    if geo
+      # TODO? This works. Is there a more correct (railsy) way of doing this
+      # e.g. using write_attribute ? What is the trade off?
+      self.longitude= geo['longitude']
```

Comment:
don't think you need `self` in these, looks fine to me tho!

Reply:
Ahhhhh IIRC now I think you do need self if you want to write (as opposed to read). I don't think I'm going totally mad!